### PR TITLE
Collection of small fixes

### PR DESCRIPTION
**Pull request by paulromano**
This PR is a potpourri of small code and documentation fixes that I've identified over the last few weeks.  To summarize:

- There are several documentation fixes (formula for Maxwell distribution, instructions in cross section configuration)
- Angle distributions in our data class hierarchy are stored with polymorphic pointers, but in fact they are always represented as `Tabular` objects. I've changed from a generic `std::unique_ptr<Distribution>` to `std::unique_ptr<Tabular>`.
- There was a bug in reading discrete photon energy distributions from ENDF files (never used for anything "real")
- The only real bug fix here is explained in #1704. When the "nearest" temperature method is used and the tolerance is very high, the S(a,b) data ends up using a temperature that may not actually be the nearest. This should be fixed now.

### CODE REVIEW

**Comment by paulromano** [comment]
Thanks for the suggestion @gridley — just added that

File: src/thermal.cpp [hunk 1]
```diff
@@ -151,25 +151,25 @@ ThermalScattering::calculate_xs(double E, double sqrtkT, int* i_temp,
...
-    // If using nearest temperature, do linear search on temperature
-    for (i = 0; i < kTs_.size(); ++i) {
-      if (std::abs(kTs_[i] - kT) < K_BOLTZMANN*settings::temperature_tolerance) {
-        break;
-      }
-    }
-  } else {
+  int i = 0;
+
+  if (kTs_.size() > 1) {
```
Comment (pshriwise) [inline src/thermal.cpp:156]: Correct me if I'm wrong, but I think there is still a problem here for temperatures that are lower than the set of loaded cross section temperatures. For example, if there is a model temperature at 150 K and loaded cross sections at 200 K, 500 K, 800 K, and 900K then the entire loop below will run to the end of the loaded cross sections and we'll end up with and index indicating that our temperature is between 800 K and 900 K. The temperature _farthest_ from 150 K will be used based on the selection in the next section. From what I can tell this works just fine for temperatures higher than the highest loaded cross section however.

We'll want to add a check for temps below `kTs_[0]` and skip the loop.

File: src/thermal.cpp [hunk 2]
```diff
@@ -151,25 +151,25 @@ ThermalScattering::calculate_xs(double E, double sqrtkT, int* i_temp,
...
     // Find temperatures that bound the actual temperature
     for (i = 0; i < kTs_.size() - 1; ++i) {
```
Comment (pshriwise) [inline src/thermal.cpp:158]: Correcting myself in the above comment, there _is_ a problem for temperatures above the loaded cross section range. `i` gets incremented one more time upon leaving the loop below, referencing an invalid interval in the `kTs_` vector. Updating the condition here ensures we use an index for the correct temperature interval for temperatures either in or above the highest interval and lets us skip an unnecessary loop iteration.

```suggestion
    for (i = 0; i < kTs_.size() - 1; ++i) {
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
Thanks for these updates @paulromano! Don't know how you find the time... :star_struck: A couple of comments on the updated temperature selection scheme.

File: src/thermal.cpp (hunk 2 above)
Comment (paulromano) [reply src/thermal.cpp:158]: Using a `while` loop would solve this and the problem above -- I'll fix this. Thanks for the careful review!

**Review by pshriwise — COMMENTED** [review]
Looks good @paulromano! Thanks!

**Review by pshriwise — APPROVED** [review]

**Pull request merged**